uw-frame-components/portal/notifications/services.js

```diff
@@ -4,64 +4,186 @@ define(['angular', 'jquery'], function(angular, $) {
 
   var app = angular.module('portal.notifications.services', []);
 
-  app.factory('notificationsService', ['$q','$http', 'miscService', 'PortalGroupService', 'keyValueService','SERVICE_LOC', 'KV_KEYS', function($q, $http, miscService, PortalGroupService, keyValueService, SERVICE_LOC, KV_KEYS) {
+  app.factory('notificationsService', ['$q','$http', '$log', '$filter', 'miscService', 'PortalGroupService', 'keyValueService','SERVICE_LOC', 'KV_KEYS', function($q, $http, $log, $filter, miscService, PortalGroupService, keyValueService, SERVICE_LOC, KV_KEYS) {
     // LOCAL VARIABLES
     var filteredNotificationPromise;
     var dismissedPromise;
-    var dismissedIds = [];
 
     /////////////////////
     // SERVICE METHODS //
     /////////////////////
+    
     /**
      * Get all notifications at the given URL (as set in app-config.js or override.js)
      * @returns {*} A promise which returns an object containing notifications arrays
      */
     var getAllNotifications = function() {
-      return $q.all([$http.get(SERVICE_LOC.notificationsURL, {cache : true}), getDismissedNotificationIds()])
-          .then(function(result) {
-            return sortNotifications(result[0].data.notifications, result[1]);
-          },
-          function(reason) {
-            miscService.redirectUser(reason.status, 'notifications json feed call');
-          }
-        );
+      return $http.get(SERVICE_LOC.notificationsURL, {cache : true}).then(
+        function(result) {
+          return separateNotificationsByDismissal(result.data.notifications).then(
+            function(results){
+              return results;
+            },function(reason){
+              $log.warn("Error Retrieving all notifications");
+            }
+          );
+        },
+        function(reason) {
```

Comment:
:art: Could using `.then` with `.catch` be more readable than using `.then()` with the second param catch?